lib/class/api.class.php

```diff
@@ -2357,4 +2357,107 @@ public static function democratic($input)
             break;
         } // switch on method
     } // democratic
+
+    /**
+     * ad_hoc_playlist
+     * MINIMUM_API_VERSION=400001
+     *
+     * Return songs based on supplied criteria
+     *
+     * @param array $input
+     * $input = array(mode            = (string)  "least_recent_played"|"most_recent_played"|"random"   // optional, default = "least_recent_played"
+     *                filter          = (string)  $filter                                               // optional, LIKE matched to song title
+     *                album_id        = (integer) $album_id                                             // optional
+     *                artist_id       = (integer) $artist_id                                            // optional
+     *                flagged_only    = (integer) 0|1                                                   // optional, default = 0
+     *                format          = (string)  "song"|"index"|"id"                                   // optional, default = "song"
+     */
+    public static function ad_hoc_playlist($input)
+    {
+        // parameter defaults
+        if (!in_array($input[mode], array("least_recent_played", "most_recent_played", "random"), true)) {
+            $input[mode] = "least_recent_played";
+        }
+        if (!in_array($input[format], array("song", "index", "id"), true)) {
+            $input[format] = "song";
+        }
+        
+        // process parameters
+
+        $select       = array("song.id AS song_id");
+        $from         = array("song");
+        $where        = array();
+        $order        = array();
+        $bound_values = array();
+
+        if (in_array($input[mode], array("least_recent_played", "most_recent_played"), true)) {
```

Comment:
$input[mode] should be ($input['mode']

The input array is like a dictionary

Reply:
Ah, hard to switch my brain off of perl... But of course you're right.